systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/KafkaRollerST.java

```diff
@@ -322,6 +326,152 @@ void testKafkaPodPendingDueToRack(ExtensionContext extensionContext) {
         KafkaUtils.waitForKafkaDeletion(namespaceName, clusterName);
     }
 
+    /**
+     * @description This test case examines the behavior of Kafka rollers under various node pool configurations, focusing on how rolling updates are triggered or prevented in different scenarios.
+     *
+     * @steps
+     *  1. - Initialize test environment and create node pools with different roles (mixed, broker, controller) and varying replica counts.
+     *     - Create Kafka instance with a single replica and deploy mixed node pool 'A' with 1 replica.
+     *  2. - Deploy two additional broker node pools 'B1' and 'B2' with 1 and 2 replicas, respectively.
+     *     - Verify that the creation of these broker nodes does not trigger a rolling update in the existing node pool 'A'.
+     *  3. - In KRaft mode, deploy two controller node pools 'C1' and 'C2' with 2 and 1 replicas, respectively.
+     *     - Expect a rolling update for mixed-role and broker-role nodes, but not for the controller nodes initially.
+     *  4. - After creating controller node pools, verify rolling updates across different node pools as they adjust to the new configuration.
+     *     - Check that the controller nodes roll only when their specific pool is modified.
+     *  5. - Change controller-only configuration and verify that only mixed and controller node pools are affected, while broker nodes remain unchanged.
+     *  6. - Modify broker-only configuration and observe that only mixed-role and broker-role node pools roll, leaving controller-role nodes unaffected.
+     *
+     * @usecase
+     *  - kafka-node-pool-rolling-update
+     *  - kafka-roller
+     *  - kafka-configuration-management
+     */
+    @ParallelNamespaceTest
+    void testKafkaRollerBehaviorUnderVariousScenarios(ExtensionContext extensionContext) {
+        final TestStorage testStorage = new TestStorage(extensionContext);
+        final String nodePoolNameA = testStorage.getKafkaNodePoolName() + "-a";
+        final String nodePoolNameB1 = testStorage.getKafkaNodePoolName() + "-b1";
+        final String nodePoolNameB2 = testStorage.getKafkaNodePoolName() + "-b2";
+        final String nodePoolNameC1 = testStorage.getKafkaNodePoolName() + "-c1";
+        final String nodePoolNameC2 = testStorage.getKafkaNodePoolName() + "-c2";
+
+        final int mixedPoolAReplicas = 1, brokerPoolB1Replicas = 1, brokerPoolB2Replicas = 2, controllerPoolC1Replicas = 2, controllerPoolC2Replicas = 1;
+
+        final Kafka kafka = KafkaTemplates.kafkaPersistent(testStorage.getClusterName(), 1, 1).build();
+        final KafkaNodePool mixedPoolA = KafkaNodePoolTemplates.kafkaBasedNodePoolWithDualRole(nodePoolNameA, kafka, mixedPoolAReplicas).build();
```

Comment:
this part will fail in case we will run this test without `+UseKRaft` FG.
As this would produce Kafka (KafkaNodePool) with role `Controller` , therefore failing on something like: 
```
In a ZooKeeper-based Kafka cluster, the KafkaNodePool role has to be always set only to the 'broker' role   
```

Possible solutions are: 

- use instead `kafkaBasedNodePoolWithFgBasedRole` for this KNP (giving it role broker in any case, and adding controller as well in case kraft is enabled) 
- or make test run only with kraft, 
- or replace `KNP a` with something like   2 KNP `initial broker` and `initial controller`  (with controller being created only if kraft enabled)

Reply:
> use instead kafkaBasedNodePoolWithFgBasedRole for this KNP (giving it role broker in any case, and adding controller as well in case kraft is enabled)
I am more favour in of this approach :)